### ISSUE TITLE
Ballerina Test Groups not Working When Compiler Plugin Tests Exist

### ISSUE THREAD

**Issue by ThisaruGuruge** [ballerina-platform/plugin-gradle#16]
**Description:**
When a Ballerina package has a `<package_name>-compiler-plugin` and a `<package_name>-compiler-plugin-test` modules in the gradle project, the `<package_name>-compiler-plugin-test` gradle module has a dependency on `<package_name>-ballerina:build` task.

This results in invoking `bal build` instead of `bal test` in the ballerina build task, since there is a requirement of `<package_name>-ballerina:build` task. Since the `bal build` command does not support filtering the tests by group name, we cannot provide a test groups filter to that. 

Therefore, this leads to the `-Pgroups=<>` parameter not working in the Gradle build. i.e. Even if we provide the following command, the build runs all the tests.

```shell
./gradlew test -Pgroups=test_group
```

A workaround is to skip the `compiler-plugin-tests:test` task (as shown below). But ideally, we should handle this in a better way.

```shell
./gradlew test -Pgroups=test_group -x <package_name>-compiler-plugin-tests
```

### Affected Versions
`0.9.0`

**Comment by ThisaruGuruge** [comment]
Is this fix available now? I am still seeing this issue in 0.9.0

**Comment by ThisaruGuruge** [comment]
Even the workaround seems to be not working now.

**Comment by ThisaruGuruge** [comment]
Re-opening since this seems like a blocker.

**Comment by ThisaruGuruge** [comment]
Fixed with `0.9.1` version
